File: CS/CustomReportParameterType/Controllers/HomeController.cs
using CustomReportParameterType.Services;
using DevExpress.XtraReports.Native;
using DevExpress.XtraReports.UI;
using System.Web.Mvc;

namespace CustomReportParameterType.Controllers
{
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Designer() {
            var report = new XtraReport();
            report.Extensions[SerializationService.Guid] = CustomDataSerializer.Name;
            return View(report);
        }
        public ActionResult Viewer() {
            var report = new XtraReport1();
            report.Extensions[SerializationService.Guid] = CustomDataSerializer.Name;
            return View(report);
        }
    }
}

File: VB/CustomReportParameterType/Controllers/HomeController.vb
#Region "usings"
Imports CustomReportParameterType.Services
Imports DevExpress.XtraReports.Native
Imports DevExpress.XtraReports.UI
Imports System.Web.Mvc
#End Region ' usings

Namespace CustomReportParameterType.Controllers
	#Region "#controller_Home"
	Public Class HomeController
		Inherits Controller

		#End Region
		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function Designer() As ActionResult
			Dim report = New XtraReport()
			report.Extensions(SerializationService.Guid) = CustomDataSerializer.Name
			Return View(report)
		End Function
		#Region "action_Viewer"
		Public Function Viewer() As ActionResult
			Dim report = New XtraReport1()
			report.Extensions(SerializationService.Guid) = CustomDataSerializer.Name
			Return View(report)
		End Function
		#End Region
	End Class
End Namespace
